llvm/lib/Target/Mips/MipsSEFrameLowering.cpp

```diff
@@ -432,19 +432,61 @@ void MipsSEFrameLowering::emitPrologue(MachineFunction &MF,
   MachineModuleInfo &MMI = MF.getMMI();
   const MCRegisterInfo *MRI = MMI.getContext().getRegisterInfo();
 
-  // Adjust stack.
-  TII.adjustStackPtr(SP, -StackSize, MBB, MBBI);
+  const std::vector<CalleeSavedInfo> &CSI = MFI.getCalleeSavedInfo();
+
+  int64_t CalleeSavedStackSize;
+  int64_t LoaclStackSize;
+  // If we have two-step stack setup MBBI_2 will point to the
+  // first instruction after calle-saves store sequence
+  MachineBasicBlock::iterator MBBI_2 = MBBI;
+
+  if (MipsFI->isTwoStepStackSetup(MF)) {
+
+    CalleeSavedStackSize = MipsFI->getCalleeSavedStackSize();
+    unsigned NumOfCSI = MFI.getCalleeSavedInfo().size();
+
+    // Move MBBI_2 to point to the first instruction after
+    // calle-saves store sequence. That's the place for the second
+    // steck pointer adjustment.
+    std::advance(MBBI_2, NumOfCSI);
```

Comment:
Yes, I think that's the safer way. However, in order to do that I think we need to introduce a new function which checks opcodes and operands of skipped instructions. This function will be nanomips-specific  so I think it's not a good idea to write it inside ```MipsSEFrameLowering```.  Maybe a good place to write such function is inside ```MipsFunctionInfo``` since we have already put ```CalleeSavedStackSize``` variable there? Is this a good approach for this problem?  

Note that on the line [494](https://github.com/MediaTek-Labs/llvm-project/pull/1/files#diff-2436c22f93c462e0e51bcd9e774e00bfcd69bac4b0246db61fe28a1f796b8397R494) we already had this kind of skipping CSR store instructions.